File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int l=0;
        int r=arr.length-1;
        int prev=-1;
        for(int i=r;i>=l;i--){
            int cur=arr[i];
            arr[i]=prev;
            if(cur>prev){
                prev=cur;
            }
        }
        return arr;
    }
}

File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* replaceElements(int* arr, int arrSize, int* returnSize) {
    *returnSize=arrSize;
    int high=-1;
        for(int i=arrSize-1;i>=0;i--){
            if(arr[i]>high){
                int t=arr[i];
                arr[i]=high;
                high=t;
            }else{
                arr[i]=high;
            }
        }
        return arr;
}